File: react-native-harmony/harmony/cpp/RNOHCorePackage/TurboModules/DeviceEventManagerTurboModule.cpp
#include "RNOHCorePackage/TurboModules/DeviceEventManagerTurboModule.h"

namespace rnoh {
using namespace facebook;

DeviceEventManagerTurboModule::DeviceEventManagerTurboModule(const ArkTSTurboModule::Context ctx, const std::string name) : ArkTSTurboModule(ctx, name) {
    methodMap_ = {
        ARK_METHOD_METADATA(invokeDefaultBackPressHandler, 0),
    };
}

} // namespace rnoh

File: tester/harmony/rnoh/src/main/ets/RNOHCorePackage/turboModules/DeviceEventManagerTurboModule.ts
import { TurboModule } from "../../RNOH/TurboModule";

export class DeviceEventManagerTurboModule extends TurboModule {
  public static readonly NAME = 'DeviceEventManager';

  invokeDefaultBackPressHandler() {
    this.ctx.uiAbilityContext.terminateSelf();
  }
}
